components/src/main/java/uk/gov/hmrc/components/organism/information/InformationMessageCardView.kt

```diff
@@ -92,26 +92,45 @@ class InformationMessageCardView @JvmOverloads constructor(
         }
     }
 
-    fun setHeadlineButtons(buttons: List<SecondaryButton>) {
+    fun setHeadlineButtons(buttons: List<InformationMessageButton>) {
         binding.headlineButtonsContainer.removeAllViews()
 
         val spacing8 = context.resources.getDimensionPixelSize(R.dimen.hmrc_spacing_8)
         val spacing16 = context.resources.getDimensionPixelSize(R.dimen.hmrc_spacing_16)
 
         binding.headlineButtonsContainer.visibility = if (buttons.isNotEmpty()) View.VISIBLE else View.GONE
         buttons.forEach {
-            it.minHeight = 0
-            it.setMargins(spacing16, spacing8, spacing16, 0)
-            it.backgroundTintList = ContextCompat.getColorStateList(context, R.color.hmrc_white)
-            it.gravity = Gravity.CENTER
-            binding.headlineButtonsContainer.addView(it)
+            it.button.apply {
+                minHeight = 0
+                setMargins(spacing16, spacing8, spacing16, 0)
+                gravity = Gravity.CENTER
+            }
+            if (it.isOutlineButton) {
+                it.button.apply {
+                    strokeWidth = OUTLINE_STROKE_WIDTH
+                    strokeColor = ContextCompat.getColorStateList(context, type!!.headlineTint)
+                    setTextColor(ContextCompat.getColorStateList(context, type!!.headlineTint))
+                }
+            } else {
+                it.button.backgroundTintList = ContextCompat.getColorStateList(context, R.color.hmrc_white)
+            }
+            binding.headlineButtonsContainer.addView(it.button)
         }
     }
 
+    data class InformationMessageButton(
```

Comment:
It might be easier for callers if this was put into its own class. At the moment they will need to write `InformationMessageCardView.InformationMessageButton()`. If you extract it out then they can call `InformationMessageButton()` directly. Of course they could import the longer version but extracting this data class would make it a bit tidier.